### ISSUE TITLE
[enhancement] search/display results differently

### ISSUE THREAD

**Issue by jghauser** [nvim-telescope/telescope-bibtex.nvim#8]
First of all, I want to thank you for making this excellent extension to telescope and for responding so super quickly to the issues that I reported before. I've now started using telescope-bibtex as my main citation inserter and it works very well!

I was thinking that it could be even more useful, if instead of searching/displaying the citation key, we could search/display something like `<author> <year> <title>`. I think most people don't know their citation keys very well, and they might not find the required entries because the word they're searching isn't in the citation key but only in the title. Displaying things properly might be a little complicated (adjusting for different widths, long strings, multiple authors...), but at least searching all values of the bibtex entry rather than just the citation key would be very helpful (well, actually I don't know if that's any easier to implement).

Thanks again!

**Comment by noahares** [comment]
I see what you mean and indeed, this would be really helpful to have. After thinking about it, this might be easier done than I initially thought because all I would need to do is extract all values (the part after the "=") and tweak the get_entry function I already have for the preview. Then one could search everything (maybe restricted in some way by user-defined keys like `searchable_entries = {"author", "year", "title"}` and still have the whole entry in preview. The knick would be to get the correct entry after selecting the searched item, but that should be possible via the `value`, `display` etc keys of `entry_maker` :)

Ignore my loud thinking while I try to write things down here so I don't forget them.
TLDR: Great idea! I am polishing my commenter plugin right now, so this has prio. But afterwards I will see what I can do :)

**Comment by jghauser** [comment]
Great, looking forward to seeing this implemented! Happy to test once it's done :)

**Comment by noahares** [comment]
Hey, quick update: I still got the feature on my radar, but I am busy until at least this weekend. Might get something done then.

**Comment by jghauser** [comment]
Thanks for the update! No problem of course -- you're working on this in your spare time, and I'm very happy having something that works pretty well already for my use case! :)

**Comment by kofm** [comment]
This would be an awesome enhancement to this very very very useful plugin 
Thanks for this awesome work

**Comment by noahares** [comment]
Thanks for reminding me about it! I totally forgot. I will go over the branch again and merge it into master.

**Comment by kofm** [comment]
Thank you very much! I tried it for a while, can't live without it anymore :D

**Comment by noahares** [comment]
I just pushed some improvements to the branch. Let me know if any bugs persist. If I dont get any complaints during the next few days I will merge it.
